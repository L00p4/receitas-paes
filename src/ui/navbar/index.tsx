import { WrapperNavbar } from './navbar.styles'

export { default as NavbarSkeleton } from './navbar.skeleton'

const Navbar = () => (
  <WrapperNavbar>
    <div>
      <h1>Meu pão na sua mão</h1>
    </div>
    <div>
      <ul>
        <li>Receitas</li>
      </ul>
    </div>
  </WrapperNavbar>
)

export default Navbar
