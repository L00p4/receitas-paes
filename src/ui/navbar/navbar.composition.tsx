import React from 'react'

import Navbar, { NavbarSkeleton } from '.'

export const Primary = () => <Navbar />

export const Skeleton = () => <NavbarSkeleton />
