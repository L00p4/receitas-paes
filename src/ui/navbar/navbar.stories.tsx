import { Meta, StoryObj } from '@storybook/react'

import Navbar, { NavbarSkeleton } from '.'

const meta: Meta<typeof Navbar> = {
  title: 'Navbar',
  component: Navbar,
  tags: ['autodocs'],
  args: {}
}

type Story = StoryObj<typeof Navbar>

export const Primary: Story = {
  args: {}
}

export const Skeleton: Story = {
  render: () => <NavbarSkeleton />,
  args: {}
}

export default meta
