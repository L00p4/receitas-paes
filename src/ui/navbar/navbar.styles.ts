import styled from 'styled-components'

export const WrapperNavbar = styled.nav`
  display: flex;
  justify-content: space-around;

  div:nth-child(1) {
    font-family: 'Shadows Into Light', sans-serif;
  }

  div:nth-child(2) {
    li {
      list-style-type: none;
    }
  }
`
