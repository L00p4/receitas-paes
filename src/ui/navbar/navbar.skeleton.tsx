import React from 'react'

import { WrapperNavbar } from './navbar.styles'

const NavbarSkeleton = () => (
  <WrapperNavbar>
    <h1 className="skeleton">Navbar</h1>
  </WrapperNavbar>
)

export default NavbarSkeleton
