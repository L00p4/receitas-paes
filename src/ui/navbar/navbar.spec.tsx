import '@testing-library/jest-dom'
import 'jest-environment-jsdom'
import 'jest-styled-components'

import { render, screen } from '@testing-library/react'

import Navbar from '.'
import { Primary, Skeleton } from './navbar.composition'

describe('<Navbar />', () => {
  it('should render the heading', () => {
    render(<Navbar />)

    expect(
      screen.getByRole('heading', { name: /meu pão na sua mão/i })
    ).toBeInTheDocument()
  })

  it('should render Primary', () => {
    expect(render(<Primary />).container).toMatchSnapshot()
  })

  it('should render Skeleton', () => {
    expect(render(<Skeleton />).container).toMatchSnapshot()
  })
})
