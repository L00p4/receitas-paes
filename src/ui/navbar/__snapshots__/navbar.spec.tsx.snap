// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render Primary 1`] = `
.c0 {
  display: flex;
  justify-content: space-around;
}

.c0 div:nth-child(1) {
  font-family: 'Shadows Into Light',sans-serif;
}

.c0 div:nth-child(2) li {
  list-style-type: none;
}

<div>
  <nav
    class="c0"
  >
    <div>
      <h1>
        Meu pão na sua mão
      </h1>
    </div>
    <div>
      <ul>
        <li>
          Receitas
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`<Navbar /> should render Skeleton 1`] = `
.c0 {
  display: flex;
  justify-content: space-around;
}

.c0 div:nth-child(1) {
  font-family: 'Shadows Into Light',sans-serif;
}

.c0 div:nth-child(2) li {
  list-style-type: none;
}

<div>
  <nav
    class="c0"
  >
    <h1
      class="skeleton"
    >
      Navbar
    </h1>
  </nav>
</div>
`;
